### DISCUSSION TITLE
Lesson 10: AAVE v3 Did anyone get WETHgateway.depositETH() to work?

### DISCUSSION THREAD

**Discussion by n4n0b1t3** [smartcontractkit/full-blockchain-solidity-course-py#1327] [Q&A]
_edit: I drastically shortened the post because it was a little confusing and became way to long._

## My goal: AAVE v3, deposit ETH and borrow DAI
I played around with AAVE in lesson 10 and as long as I stick to the course and use the v2 code provided in the lesson it works just fine. So, I tried to switch to v3 just to see if I can apply the concept and understand the documentation. I did not get it to work. This is just a call out if any of you managed to get this to work and would be willing to share the code.

**Comment by cromewar** [comment]
This question got me out of the blue, could be give a hand on this one please @PatrickAlphaC ?

**Comment by PatrickAlphaC** [comment]
Hey @n4n0b1t3! This is a lot to triage with you... I love how you're going above and beyond, (and this is what you SHOULD do, so kudos to you!) but it's a lot to ask us to help debug the new stuff you work on.

From my quick two cents, it's likely you have a wrong address, hence why it's reverting all you.

Good luck!

**Reply by n4n0b1t3** [reply]
Hi @PatrickAlphaC, I am sorry, this was a  misunderstanding, I didn't ask for debugging, just if someone had done this before and is willing to share. Cromewar added you to the conversation. I learned a lot about AAVE version two now and believe I can soon apply this to v3.

**Reply by PatrickAlphaC** [reply]
Gotcha, looking for a hand I see, excellent! 

It's not something I've done before, sorry.

**Comment by alexkalen** [comment]
Im also interested in learning how Aave V3 works! If you have any new input on this I would appreciate it.
